solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,495 @@
+    api      = optional(list(string), [])
+  })
+  default = {}
+}
+
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "private"
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are `public`, `private`, or `public-and-private`."
+  }
+}
+
+##############################################################
+# KMS Related
+##############################################################
+variable "kms_encryption_enabled_cluster" {
+  description = "Set to true to enable KMS encryption on the Object Storage bucket created for the Security and Compliance Center instance. When set to true, a value must be passed for either `existing_cluster_kms_key_crn` or `existing_kms_instance_crn` (to create a new key). Can not be set to true if passing a value for `existing_scc_instance_crn`."
+  type        = bool
+  default     = false
+  nullable    = false
+
+  validation {
+    condition     = var.existing_kms_instance_crn != null ? var.kms_encryption_enabled_cluster : true
+    error_message = "If passing a value for 'existing_kms_instance_crn', you should set 'kms_encryption_enabled_cluster' to true."
+  }
+
+  validation {
+    condition     = var.existing_cluster_kms_key_crn != null ? var.kms_encryption_enabled_cluster : true
+    error_message = "If passing a value for 'existing_cluster_kms_key_crn', you should set 'kms_encryption_enabled_cluster' to true."
+  }
+
+  validation {
+    condition     = var.kms_encryption_enabled_cluster ? ((var.existing_cluster_kms_key_crn != null || var.existing_kms_instance_crn != null) ? true : false) : true
+    error_message = "Either 'existing_cluster_kms_key_crn' or 'existing_kms_instance_crn' is required if 'kms_encryption_enabled_cluster' is set to true."
+  }
+}
+
+variable "existing_kms_instance_crn" {
+  type        = string
+  default     = null
+  description = "The CRN of an existing KMS instance (Hyper Protect Crypto Services or Key Protect). Used to create a new KMS key unless an existing key is passed using the `existing_scc_cos_kms_key_crn` input. If the KMS instance is in different account you must also provide a value for `ibmcloud_kms_api_key`."
+
+  validation {
+    condition = anytrue([
+      can(regex("^crn:(.*:){3}(kms|hs-crypto):(.*:){2}[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}::$", var.existing_kms_instance_crn)),
+      var.existing_kms_instance_crn == null,
+    ])
+    error_message = "The provided KMS instance CRN in the input 'existing_kms_instance_crn' in not valid."
+  }
+}
+
+variable "force_delete_kms_key" {
+  type        = bool
+  default     = false
```

Comment:
Oh this doesn't cause any issue like it does with secrets manager? Did you do a test to make sure the KMS key destroys successfully when this is set to false? You currently have no PR tests that are testing key creation, so you might hit it after updating the test

Reply:
Changed the default value to `true`